graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/user/TopSecondDegreeByCountForUser.java

```diff
@@ -48,21 +49,26 @@ public TopSecondDegreeByCountForUser(
 
   @Override
   protected void updateNodeInfo(
+    TopSecondDegreeByCountRequestForUser request,
     long leftNode,
     long rightNode,
     byte edgeType,
     double weight,
-    EdgeIterator edgeIterator,
-    int maxSocialProofTypeSize) {
+    EdgeIterator edgeIterator) {
     NodeInfo nodeInfo;
 
+    long segmentCreationTime = ((TimestampEdgeIterator)edgeIterator).currentSegmentCreationTime();
```

Comment:
"segment" is an implementation detail, and it is better not expose it through the interface. you can name the interface  method as getCurrentEdgeEngagementTime().